### PR TITLE
 [Backport] Fixed subscribe to newsletter if you already have an account issue

### PR DESCRIPTION
**Pull request by ravi-chandra3197**
### Description (*)

Fixed subscribe to newsletter if you already have an account issue
### Fixed Issues (if relevant)

1. magento/magento2#8952: You can't subscribe to newsletter if you already have an account

### Manual testing scenarios (*)

N/A

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Newsletter/Controller/Subscriber/NewAction.php
Comment (sidolov) [inline app/code/Magento/Newsletter/Controller/Subscriber/NewAction.php]: Please, combine two `if` statements to one

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by ravi-chandra3197** [comment]
Hello @sidolov 
I have update change as per your suggestion can review it.

**Comment by ravi-chandra3197** [comment]
Hello @sidolov
can you guide how to add the custom test case in magento2 like my Travis fail due to default test case I need to add custom test case so can you guide

**Comment by ravi-chandra3197** [comment]
Hello @sidolov
I have update change as per your suggestion can review it.

**Review by orlangur — CHANGES_REQUESTED** [review]
Nowadays all changes must be applied to `2.3-develop` first (see https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#rules). Please prepare a new pull request, this one should be `on hold` until the latter is merged.

**Comment by ravi-chandra3197** [comment]
Fixed subscribe to newsletter if you already have an account issue #8952 -2.3 develop #19164

**Comment by ravi-chandra3197** [comment]
#19164 Fixed subscribe to newsletter if you already have an account issue #8952 -2.3 develop #19164

**Comment by orlangur** [comment]
@ravi-chandra3197 after 2.3 PR is merged, please rewrite history of this branch to reflect it (cherry pick the only meaningful commit).

**Comment by ravi-chandra3197** [comment]
Hello, @orlangur I have rewrite branch history with single commit can review it.

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-3680](https://jira.corp.magento.com/browse/ENGCOM-3680) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197. Thank you for your contribution. 
We will aim to release these changes as part of 2.2.8. 
Please check the release notes for final confirmation.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#porting-tool) to port commits automatically.

**Pull request merged**
